src/metabase/search/scoring.clj

```diff
@@ -164,14 +183,13 @@
   (/ (or (model->sort-position model) 0)
      (count model->sort-position)))
 
-(defn- text-score-with-match
+(defn- text-scores-with-match
   [raw-search-string result]
   (if (seq raw-search-string)
-    (text-score-with match-based-scorers
-                     (tokenize (normalize raw-search-string))
-                     result)
-    {:score  0
-     :match  ""}))
+    (text-scores-with match-based-scorers
+                      (tokenize (normalize raw-search-string))
+                      result)
+    [{:score 0 :weight 1 :match ""}]))
```

Comment:
I'm wondering if `text-scores-with` should handle the nil `raw-search-string` case so that its `{:score 0 ...}` result can be returned if there's no search string or if there's no match, instead of having the separate `{:score 0 ...}` here.

I also don't understand why the one in `text-scores-with` has weight 0 and this one has weight 1.

(I have a sinking sensation that if you `git blame` long enough you'll discover I'm the one who wrote it that way in the first place. If so, sorry!)

Reply:
Good catch, I'll try having `text-scores-with-match` handle the empty `raw-search-string`. I also have the philosophy that the more stuff someone does the more mistakes they'll make so mistakes aren't such a bad thing! Mine or yours!